### PR TITLE
Functions for Handling Text vs Binary Messages on WS

### PR DESCRIPTION
**Pull request by dvonthenen**
This implements various different mechanisms for sending data... whether it's in a queue or it's immediately (bypassing the queue) direct with a byte buffer.

Tested using the `Send()` and `SendImmedaitely()` using the microphone example. Works just fine.

## Summary by CodeRabbit

- **Refactor**
	- Improved thread safety and message handling in live client operations through enhanced locking mechanisms and distinct methods for binary and text messages.

### CODE REVIEW

File: Deepgram/Clients/Live/v1/Client.cs [hunk 1]
```diff
@@ -18,7 +19,9 @@ public class Client : Attribute, IDisposable
 
     private ClientWebSocket? _clientWebSocket;
     private CancellationTokenSource? _cancellationTokenSource;
-    private readonly Mutex _mutex = new Mutex();
+
+    private readonly Mutex _mutexSubscribe = new Mutex();
+    private readonly Mutex _mutexSend = new Mutex();
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs]: will do

File: Deepgram/Clients/Live/v1/Client.cs [hunk 2]
```diff
@@ -171,7 +174,7 @@
     /// <returns>True if successful</returns>
     public bool Subscribe(EventHandler<OpenResponse> eventHandler)
     {
-        lock (_mutex)
+        lock (_mutexSubscribe)
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:177]: I hit a compile error depending on the version of .NET. Ignoring suggestion.

File: Deepgram/Clients/Live/v1/Client.cs [hunk 3]
```diff
@@ -261,21 +264,59 @@
...
+    /// This method sends a text message over the WebSocket connection immediately without queueing.
+    /// </summary>
+    public void SendMessageImmediately(byte[] data)
+    {
+        lock (_mutexSend)
+        {
+            _clientWebSocket.SendAsync(new ArraySegment<byte>(data), WebSocketMessageType.Text, true, _cancellationTokenSource.Token)
+                .ConfigureAwait(false);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs]: This is already separated out.

File: Deepgram/Clients/Live/v1/Client.cs [hunk 4]
```diff
@@ -309,7 +350,11 @@
                 while (_sendChannel.Reader.TryRead(out var message))
                 {
                     Log.Verbose("ProcessSendQueue", $"Sending message..."); // TODO: dump this message
-                    await _clientWebSocket.SendAsync(message.Message, message.MessageType, true, _cancellationTokenSource.Token).ConfigureAwait(false);
+                    lock (_mutexSend)
+                    {
+                        _clientWebSocket.SendAsync(message.Message, message.MessageType, true, _cancellationTokenSource.Token)
+                            .ConfigureAwait(false);
+                    }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:359]: This should never deadlock. Dismissing suggestion.

File: Deepgram/Clients/Live/v1/Client.cs [hunk 5]
```diff
@@ -350,9 +395,12 @@
...
                 Log.Debug("ProcessKeepAlive", "Sending KeepAlive");
-                byte[] array = Encoding.ASCII.GetBytes("{\"type\": \"KeepAlive\"}");
-                await _clientWebSocket.SendAsync(new ArraySegment<byte>(array), WebSocketMessageType.Text, true, _cancellationTokenSource.Token)
-                    .ConfigureAwait(false);
+                byte[] data = Encoding.ASCII.GetBytes("{\"type\": \"KeepAlive\"}");
+                lock (_mutexSend)
+                {
+                    _clientWebSocket.SendAsync(new ArraySegment<byte>(data), WebSocketMessageType.Text, true, _cancellationTokenSource.Token)
+                        .ConfigureAwait(false);
+                }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:405]: This should never deadlock. Dismissing suggestion.

File: Deepgram/Clients/Live/v1/Client.cs [hunk 6]
```diff
@@ -658,8 +706,11 @@
                 Log.Debug("Stop", "Sending Close message...");
 
                 // send a close to Deepgram
-                await _clientWebSocket.SendAsync(new ArraySegment<byte>([]), WebSocketMessageType.Binary, true, cancelToken)
-                    .ConfigureAwait(false);
+                lock (_mutexSend)
+                {
+                    _clientWebSocket.SendAsync(new ArraySegment<byte>([]), WebSocketMessageType.Binary, true, cancelToken)
+                        .ConfigureAwait(false);
+                }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:715]: This should never deadlock. Dismissing suggestion.

**Review by lukeocodes — APPROVED** [review]

**Pull request merged**